apps/settings/js/modules/wifi_utils.js

```diff
@@ -187,20 +186,14 @@ define(function(require) {
       };
 
       var checkPassword = function() {
-        var isSSIDInvalid = function() {
-          if (ssid) {
-            return (ssid.value.length === 0);
-          } else {
-            return false;
-          }
-        };
-
+        var key = WifiHelper.getKeyManagement(network);
         submitButton.disabled =
           !WifiHelper.isValidInput(key, password.value, identity.value,
-            eap.value) || isSSIDInvalid();
+            eap.value) || !WifiHelper.isSSIDValid(ssid.value);
       };
 
       eap.onchange = function() {
+        var key = WifiHelper.getKeyManagement(network);
```

Comment:
The `network` object will be changed if user changes "Security" type in "Authentication" page.